api/ccapi/include/tensor_dim.h

```diff
@@ -22,16 +22,17 @@
 #include <bitset>
 #include <vector>
 
-namespace nntrainer {
-
-constexpr const size_t MAXDIM = 4;
+namespace ml {
+namespace train {
 
 /**
  * @brief Tensor Dimension. This class is used to save dimension information
  *
  */
 class TensorDim {
```

Comment:
this is minor but can you also move the implementations out of this header to its source, so that if some implementation is updated for bugfix, header remains same.

Reply:
I made a another pr for this cause this will gonna too huge.
The following pr(#1456) is related with this.